### PR TITLE
Fix stream manager get progress

### PR DESCRIPTION
**Pull request by asias**
streaming: Fix map access in stream_manager::get_progress
 
 When the progress is queried, e.g., query from nodetool netstats
 the progress info might not be updated yet.
 
 Fix it by checking before access the map to avoid errors like:
 
 std::out_of_range (_Map_base::at)
 
 Fixes: #5437
 Tests: nodetool_additional_test.py:TestNodetool.netstats_test

### CODE REVIEW

File: streaming/stream_transfer_task.cc [hunk 1]
```diff
@@ -176,13 +176,6 @@ future<> send_mutations(lw_shared_ptr<send_info> si) {
 }
```
Comment (avikivity) [inline streaming/stream_transfer_task.cc]: Why is this reverted? the changelog should explain.

File: streaming/stream_manager.cc [hunk 2]
```diff
@@ -164,11 +164,20 @@ void stream_manager::remove_progress(UUID plan_id) {
 }
 
 stream_bytes stream_manager::get_progress(UUID plan_id, gms::inet_address peer) const {
+    if (!_stream_bytes.count(plan_id)) {
```
Comment (avikivity) [inline streaming/stream_manager.cc]: A better pattern is _stream_bytes.find(), it avoids the double-check. It doesn't matter much here but it's better practice.

File: streaming/stream_transfer_task.cc (hunk 1 above)
Comment (asias) [reply streaming/stream_transfer_task.cc]: Sorry, left over during debug. I will remove it.

File: streaming/stream_manager.cc (hunk 2 above)
Comment (asias) [reply streaming/stream_manager.cc]: Yes, I agree. I will do it in v2 anyway.

**Comment by asias** [comment]
Changes in b0b5ae21696cc3be06686b6a425ad71e311fb1d9:
- Drop the revert does not belong to this series.
- Use map::find() to instead of map::count

**Comment by asias** [comment]
@avikivity Ping.

**Comment by slivne** [comment]
@avikivity ping

**Comment by slivne** [comment]
@avikivity ping

**Comment by asias** [comment]
@avikivity ping

File: streaming/stream_result_future.cc [hunk 3]
```diff
@@ -131,7 +131,7 @@ void stream_result_future::maybe_complete() {
             }
             *stats = format("tx={:d} KiB, {} KiB/s, rx={:d} KiB, {} KiB/s", sbytes.bytes_sent / 1024, tx_bw, sbytes.bytes_received / 1024, rx_bw);
         }).handle_exception([plan_id] (auto ep) {
-            sslog.warn("[Stream #{}] Fail to get progess on all shards: {}", plan_id, ep);
+            sslog.warn("[Stream #{}] Fail to get progress on all shards: {}", plan_id, ep);
```
Comment (avikivity) [inline streaming/stream_result_future.cc]: This was already merged, 6b7344f6e56

**Comment by avikivity** [comment]
Second patch queued as 145fd0313abc1fd4d7eed7d4401e2a152430bf85.

**Pull request closed without merging**